RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIPCut.cc

```diff
@@ -0,0 +1,60 @@
+#include "RecoTauTag/RecoTau/interface/TauDiscriminationProducerBase.h"
+#include <FWCore/ParameterSet/interface/ConfigurationDescriptions.h>
+#include <FWCore/ParameterSet/interface/ParameterSetDescription.h>
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/TauReco/interface/PFTauDiscriminator.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+/* 
+ * class PFRecoTauDiscriminationByIPCut
+ */
+
+using namespace reco;
+
+class PFRecoTauDiscriminationByIPCut : public PFTauDiscriminationProducerBase {
+public:
+  explicit PFRecoTauDiscriminationByIPCut(const edm::ParameterSet& iConfig)
+      : PFTauDiscriminationProducerBase(iConfig),
+        m_TausTIPToken(consumes<PFTauTIPAssociationByRef>(iConfig.getParameter<edm::InputTag>("TausIP"))),
+        m_tauTIPSelectorString(iConfig.getParameter<std::string>("Cut")),
+        m_tauTIPSelector(m_tauTIPSelectorString) {}
```

Comment:
```suggestion
        tausTIPToken_(consumes<PFTauTIPAssociationByRef>(iConfig.getParameter<edm::InputTag>("TausIP"))),
        tauTIPSelectorString_(iConfig.getParameter<std::string>("cut")),
        tauTIPSelector_(m_tauTIPSelectorString) {}
```

Starting data members with a lower case and trailing "_" is the preferred way